### PR TITLE
Tools: fix Windows path normalization and SPECIAL_SUBSYSTEMS ordering

### PR DESCRIPTION
**Pull request by ardozcnn**
### Summary

Normalizes backslashes to forward slashes in `allowed_subsystems.py` to support Windows paths, and reorders returned candidate subsystems so specific `SPECIAL_SUBSYSTEMS` take priority over generic library names.

### Classification & Testing (check all that apply and add your own)

- [x] Checked by a human programmer
- [x] Non-functional change
- [x] No-binary change
- [x] Infrastructure change (e.g. unit tests, helper scripts)
- [ ] Automated test(s) verify changes (e.g. unit test, autotest)
- [ ] Tested manually, description below (e.g. SITL)
- [ ] Tested on hardware
- [ ] Logs attached
- [ ] Logs available on request

### Description

This PR addresses two minor edge cases in `Tools/scripts/allowed_subsystems.py`:

1. **Windows Path Handling:** Added backslash (`\`) normalization to forward slashes (`/`) in `subsystems_for_path()` so that file paths passed on Windows environments do not fall through as root files.
2. **Subsystem Priority Order:** Reordered candidate subsystems returned for `SPECIAL_SUBSYSTEMS` from `[lib, sub_prefix]` to `[sub_prefix, lib]` to ensure that specific subsystem prefixes take priority over generic library directory names, matching the "most conventional first" convention documented in the script.

### CODE REVIEW

File: Tools/scripts/allowed_subsystems.py [hunk 1]
```diff
@@ -190,6 +190,7 @@ def subsystems_for_path(self, path):
         matches; callers should treat that as "needs a mapping rule".
         '''
         # normalise a leading "./"
+        path = path.replace('\\', '/')
```
Comment (peterbarker) [inline Tools/scripts/allowed_subsystems.py:193]: How do you get to here with backslashes in `path`?

File: Tools/scripts/allowed_subsystems.py [hunk 2]
```diff
@@ -218,7 +219,7 @@ def subsystems_for_path(self, path):
             filename = parts[-1]
             for sub_prefix in self.SPECIAL_SUBSYSTEMS.get(lib, []):
                 if filename.startswith(sub_prefix):
-                    return [lib, sub_prefix]
+                    return [sub_prefix, lib]
```
Comment (peterbarker) [inline Tools/scripts/allowed_subsystems.py:222]: The numbers don't support the "most conventional" motivation here.

We only grudgingly allow e.g. AC_Circle for AC_WPNav commits, for example.

**Comment by ardozcnn** [comment]
Thanks for the review, @peterbarker! I added this as a defensive check assuming Windows environment paths might pass backslashes, but if git and the internal tools always feed normalized POSIX paths here, I'm happy to remove it. Also  I misread the intended priority between the library directory name and the special prefixes. Thanks for clarifying that the library directory name is the preferred convention. I will revert this change.
